File: solutions/zig/isogram/4/isogram.zig
pub fn isIsogram(str: []const u8) bool {
    var seen: u26 = 0;

    for (str) |char| {
        const offset = switch (char) {
            'a'...'z' => char - 'a',
            'A'...'Z' => char - 'A',
            else => continue,
        };

        const candidate = @as(u26, 1) << @as(u5, @truncate(offset));
        if (candidate & seen == candidate) {
            return false;
        }

        seen |= candidate;
    }

    return true;
}

File: solutions/kotlin/isogram/4/src/main/kotlin/Isogram.kt
object Isogram {

    fun isIsogram(input: String): Boolean {
        var seen = 0
        for(c in input.lowercase()) {
            if(c.isLetter() && seen and (1 shl c.code) != 0) return false
            seen = seen or (1 shl c.code)
        }
        return true
    }
}
